tests/catch/unit/memory/MemUtils.hh

```diff
@@ -0,0 +1,638 @@
+      }
+      break;
+
+    case memType::hipMemsetD8:
+      if (async) {
+        HIP_CHECK(hipMemsetD8Async(reinterpret_cast<hipDeviceptr_t>(aPtr + data.offset), value,
+                                   count, stream));
+      } else {
+        HIP_CHECK(hipMemsetD8(reinterpret_cast<hipDeviceptr_t>(aPtr + data.offset), value, count));
+      }
+      break;
+
+    case memType::hipMemsetD16:
+      if (async) {
+        HIP_CHECK(hipMemsetD16Async(reinterpret_cast<hipDeviceptr_t>(aPtr + data.offset), value,
+                                    count, stream));
+      } else {
+        HIP_CHECK(hipMemsetD16(reinterpret_cast<hipDeviceptr_t>(aPtr + data.offset), value, count));
+      }
+      break;
+
+    case memType::hipMemsetD32:
+      if (async) {
+        HIP_CHECK(hipMemsetD32Async(reinterpret_cast<hipDeviceptr_t>(aPtr + data.offset), value,
+                                    count, stream));
+      } else {
+        HIP_CHECK(hipMemsetD32(reinterpret_cast<hipDeviceptr_t>(aPtr + data.offset), value, count));
+      }
+      break;
+
+    case memType::hipMem2D:
+      ptrOffset = getPtrOffset(data);
+      if (async) {
+        HIP_CHECK(
+            hipMemset2DAsync(aPtr + ptrOffset, data.pitch, value, data.width, data.height, stream));
+      } else {
+        HIP_CHECK(hipMemset2D(aPtr + ptrOffset, data.pitch, value, data.width, data.height));
+      }
+      break;
+
+    case memType::hipMem3D:
+      ptrOffset = getPtrOffset(data);
+      hipExtent extent;
+      extent.width = data.width;
+      extent.height = data.height;
+      extent.depth = data.depth;
+      if (async) {
+        HIP_CHECK(hipMemset3DAsync(
+            make_hipPitchedPtr(aPtr + ptrOffset, data.pitch, data.width, data.height), value,
+            extent, stream));
+      } else {
+        HIP_CHECK(
+            hipMemset3D(make_hipPitchedPtr(aPtr + ptrOffset, data.pitch, data.width, data.height),
+                        value, extent));
+      }
+      break;
+
+    default:
+      REQUIRE(false);
+      break;
+  }
+}
+
+template <typename T> static inline void freeStuff(T* aPtr, allocType type) {
+  switch (type) {
+    case allocType::deviceMalloc:
+      hipFree(aPtr);
+      break;
+    case allocType::hostMalloc:
+      hipHostFree(aPtr);
+      break;
+    default:  // for host and device registered
+      HIP_CHECK(hipHostUnregister(aPtr));
+      delete[] aPtr;
+      break;
+  }
+}
+
+/*
+ * Copies device data to host and checks that each element is equal to the
+ * specified value
+ */
+template <typename T>
+static inline void verifyData(T* aPtr, size_t value, MultiDData& data, allocType type,
+                              memType memType) {
+  auto dataH = data.height == 0 ? 1 : data.height;
+  auto dataD = data.depth == 0 ? 1 : data.depth;
+  size_t sizeInBytes = data.pitch * dataH * dataD;
+  std::unique_ptr<T[]> hostPtr = std::make_unique<T[]>(sizeInBytes);
```

Comment:
make_unique<T[]>(..) takes the number of elements as the argument, so "sizeInBytes" is a sus

Reply:
Fixed

Reply:
You're still using sizeInBytes here. I see it's defined as width*H*D, so maybe it just has the wrong name?